grammars/javascript.cson

```diff
@@ -1796,13 +1796,62 @@
           |fileoverview|final|fires|for|function|global|host|ignore|implements|inherit[Dd]oc|inner|instance|interface|kind
           |lends|license|listens|main|member|memberof|method|mixex|mixins?|module|name|namespace|nocollapse|nosideeffects
           |override|overview|package|param|preserve|private|prop|property|protected|public|read[Oo]nly|record|require[ds]
-          |returns?|see|since|static|struct|submodule|summary|template|this|throws|todo|tutorial|type|typedef|unrestricted
+          |returns?|see|since|static|struct|submodule|summary|template|this|throws|todo|type|typedef|unrestricted
           |uses|var|variation|version|virtual|writeOnce)\\b
         '''
         'name': 'storage.type.class.jsdoc'
       }
       {
         'match': '''(?x)
+          (?:
+            \\[
+              [^\\]]+            # Optional [link text] preceding {@link syntax}
+            \\]
+
+            (?!                  # Check to avoid highlighting two sets of link text
+              {
+                @\\w+            # Tagname
+                \\s+
+                [^\\s|}]+        # Namepath/URL
+                [\\s|]           # Whitespace or bar delimiting description
+                [^\\}]*
+              }
+            )
+          )?
+
+          (?:
+            {
+              (
+                @
+                (?: link         # Name of tag
+                  | linkcode
+                  | linkplain
+                  | tutorial
+                )
+              )
+              
+              \\s+
+
+              ([^\\s|}]+)        # Namepath or URL
+
+              (?:                # Optional link text following link target
+                [\\s|]           # Bar or space separating target and text
```

Comment:
The former. The latter requires a pipe: `{@link http://www.google.com|  My link}`.